### PR TITLE
[GH-13931] Return error in parseDSN func #13931

### PR DESCRIPTION
**Pull request by upwell**
Fix the issue: an error is created but not returned in the parseDNS function.

#### Summary
This PR is the fix for issue #13931: Return error in parseDSN.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13931

### CODE REVIEW

File: config/database_test.go [hunk 1]
```diff
@@ -149,6 +149,9 @@ func TestDatabaseStoreNew(t *testing.T) {
 	t.Run("invalid url", func(t *testing.T) {
 		_, err := config.NewDatabaseStore("")
 		require.Error(t, err)
+
+		_, err = config.NewDatabaseStore("mysql")
+		require.Error(t, err)
```
Comment (ali-farooq0) [inline config/database_test.go]: excellent! 👍

**Comment by hanzei** [comment]
@lindy65 A quick test that the server properly starts with mySQL and postgres should be enough

**Comment by saturninoabril** [comment]
Thanks @upwell!

**Pull request merged**